### PR TITLE
DOC: clear up warnings, fix matplotlib plot

### PR DESCRIPTION
**Pull request by mattip**
The `percentile` documentation had some misformatting that caused a failure of the  interpolation plot. I fixed that, then fixed a few more formatting errors along the way, mainly the way lists had no blank line before them.

### CODE REVIEW

**Comment by stefanv** [comment]
@mattip Did you check the rendered version of these?

File: numpy/add_newdocs.py [hunk 1]
```diff
@@ -7021,7 +7017,7 @@ def luf(lamdaexpr, *args, **kwargs):
     array(['2002-10-27T04:30:00', '2002-10-27T05:30:00', '2002-10-27T06:30:00',
            '2002-10-27T07:30:00'], dtype='<U38')
 
-    But can be made to not lose precision::
+    'casting' can be used to specify whether precision can be changed
```
Comment (mattip) [inline numpy/add_newdocs.py]: Changed content

File: numpy/core/arrayprint.py [hunk 2]
```diff
@@ -984,11 +986,12 @@ def format_float_scientific(x, precision=None, unique=True, trim='k',
...
-            k : keep trailing zeros, keep decimal point (no trimming)
-            . : trim all trailing zeros, leave decimal point
-            0 : trim all but the zero before the decimal point. Insert the
-                zero if it is missing.
-            - : trim trailing zeros and any trailing decimal point
+
+        * 'k' : keep trailing zeros, keep decimal point (no trimming)
+        * '.' : trim all trailing zeros, leave decimal point
+        * '0' : trim all but the zero before the decimal point. Insert the
+           zero if it is missing.
```
Comment (mattip) [inline numpy/core/arrayprint.py]: Formats strangely, check this

File: numpy/core/arrayprint.py [hunk 3]
```diff
@@ -1056,11 +1059,12 @@ def format_float_positional(x, precision=None, unique=True,
...
-            . : trim all trailing zeros, leave decimal point
-            0 : trim all but the zero before the decimal point. Insert the
-                zero if it is missing.
-            - : trim trailing zeros and any trailing decimal point
+
+        'k' : keep trailing zeros, keep decimal point (no trimming)
+        '.' : trim all trailing zeros, leave decimal point
+        '0' : trim all but the zero before the decimal point. Insert the
+            zero if it is missing.
+        '-' : trim trailing zeros and any trailing decimal point
```
Comment (mattip) [inline numpy/core/arrayprint.py]: This block is formatted wrongly

File: numpy/core/src/multiarray/scalartypes.c.src [hunk 4]
```diff
@@ -1607,7 +1607,8 @@ static PyGetSetDef gentype_getsets[] = {
 
 /* 0-dim array from scalar object */
 
-static char doc_getarray[] = "sc.__array__(|type) return 0-dim array";
+static char doc_getarray[] = "sc.__array__(dtype) return 0-dim array from "
+                             "scalar with specified dtype";
```
Comment (mattip) [inline numpy/core/src/multiarray/scalartypes.c.src]: Changed content

**Comment by mattip** [comment]
Yes, I checked the rendering, I left comments where I actually changed content, the rest of the changes were needed to either render correctly (lists without a leading blank line were rendered as blobs of text) or to clear up warnings

**Comment by mattip** [comment]
The `format_float_positional` and `format_float_scientific` kwarg `trim` options are better than they were but still not quite right.

File: numpy/core/code_generators/ufunc_docstrings.py [hunk 5]
```diff
@@ -3744,7 +3744,7 @@ def add_newdoc(place, name, doc):
 
 add_newdoc('numpy.core.umath', 'gcd',
     """
-    Returns the greatest common divisor of |x1| and |x2|
+    Returns the greatest common divisor of x1 and x2
```
Comment (stefanv) [inline numpy/core/code_generators/ufunc_docstrings.py]: This is a change in meaning, but I think it's still correct.

**Review by stefanv — APPROVED** [review]

File: numpy/core/arrayprint.py (hunk 3 above)
Comment (mattip) [reply numpy/core/arrayprint.py]: fixed

File: numpy/core/arrayprint.py (hunk 2 above)
Comment (mattip) [reply numpy/core/arrayprint.py]: fixed

File: numpy/core/code_generators/ufunc_docstrings.py (hunk 5 above)
Comment (mattip) [reply numpy/core/code_generators/ufunc_docstrings.py]: whoops, that should still be abs, sphinx (and me both) took it to mean a substitution. Fixing

File: numpy/doc/constants.py [hunk 6]
```diff
@@ -296,8 +296,9 @@ def add_newdoc(module, name, doc):
     See Also
     --------
     isnan : Shows which elements are Not a Number.
+
     isfinite : Shows which elements are finite (not one of
-               Not a Number, positive infinity and negative infinity)
+    Not a Number, positive infinity and negative infinity)
```
Comment (charris) [inline numpy/doc/constants.py]: Out of curiousity, where does this show up in the documentation? I cannot find it, either interactively or in the html.

Comment (eric-wieser) [reply numpy/doc/constants.py]: I think the purpose of these docs is to support:
```
import numpy.doc

help(numpy.doc.constants)
```

which strikes me as a pretty weird usage

**Comment by charris** [comment]
Thanks Matti.

**Pull request merged**